src/ansiblelint/utils.py

```diff
@@ -752,81 +752,88 @@ def get_lintables(
 
     # passing args bypass auto-detection mode
     if args:
+        matches = []
         for arg in args:
             if os.path.isdir(arg):
-                lintables.append(Lintable(arg, kind="role"))
+                for p in Path(arg).rglob("*"):
```

Comment:
The fact that we treat every folder as a role is by design, not accident. I am afraid we cannot start doing glob because it would conflict with role detection.

Example: it is invalid to have a playbookbook at the root of the role. What if user puts one there?  it means that we would not be able to decide if that folder hosts playbooks or is just a role.

Have a look at the auto-detection logic which happens when you do not pass any arguments. Likely this will produce the result you want. If it does not, look into improving it.

I do already have some plans to improve the detection.

Reply:
Ok. But how is it intended to work if someone is running `ansible-lint my_role`? E.g. the molecule playbooks are not checked right now. For a git repo git ls-files is already used to get all YAML files, so I though using a glob to achieve the same for a passed folder should be the same result.

Reply:
`get_yaml_files` is producing an OrderedDict of files to process, the idea was to produce the same OrderedDict for folders passed as args so both cases can be handled in the same way.

Reply:
When I initially implemented the auto-detection I did want to do the same and allow folders to be considered the same way.

After spending many days trying to make it work with very weird side-effects, I renounced and only did it for no-args case.

The `git ls-files` approach address more than you may think under the hood:
* allows you to call the linter from any subdirectory of your project, with same behavior
* makes use of gitignore, so you do not endup having to maintain a mirror between .gitignore and ansiblelint own config.
* allows us to have a reference root project directory, something essential as lots of things are directly dependent on it. For example requirements.yml can exist only at root level

You should also note that the only code layout official supported is the collection layout, anything not following it is dangerous and likely to cause weird bugs, sometimes even missing to identify files.

Once we improve the detection and add some testing, we ware likely to even make failure to identify a file type an error, forcing user to make a decision to either ignore or move the file. Still, this is months away as we do not want to alienate users, likely something for 5.1 or 6.0 releases.

The blocker on this is lack of tests for determine filetypes. We need to build a really big list of filenames and what they are supposed to be identified as.

Reply:
Ok

Reply:
Let's close this PR for now as it's going in a completely wrong direction?

Reply:
What I can't get out of my mind is:

> You should also note that the only code layout official supported is the collection layout

What about the legacy standalone roles layout? AFAIK it is still valid to use simple Ansible roles instead of collections, isn't it? Are there any plans from Ansible itself to completely remove standalone roles and just support collections?